src/session/content/message_row/text.rs

```diff
@@ -46,90 +50,96 @@ mod imp {
-            self.remove_css_class("outgoing");
-        }
-
-        // Show sender label, if needed
-        let show_sender = {
-            if message.is_outgoing() {
-                matches!(message.sender(), MessageSender::Chat(_))
+    fn update_widget(&self) {
+        if let Some(message) = self.message() {
+            let self_ = imp::MessageText::from_instance(self);
+
+            let content_expression = gtk::PropertyExpression::new(
+                Message::static_type(),
+                gtk::NONE_EXPRESSION,
+                "content",
+            );
+            let text_expression = gtk::ClosureExpression::new(
+                move |args| -> String {
+                    let content = args[1].get::<BoxedMessageContent>().unwrap();
+                    format_message_content_text(content.0)
+                },
+                &[content_expression.upcast()],
+            );
+
+            if let Some(message) = message.downcast_ref::<Message>() {
+                // Show sender label, if needed
+                let show_sender = {
+                    if message.is_outgoing() {
+                        matches!(message.sender(), MessageSender::Chat(_))
+                    } else {
+                        matches!(
+                            message.chat().type_(),
+                            ChatType::BasicGroup(_) | ChatType::Supergroup(_)
+                        )
+                    }
+                };
+                if show_sender {
+                    let sender_name_expression = message.sender_name_expression();
+                    sender_name_expression.bind(&*self_.sender_label, "label", gtk::NONE_WIDGET);
+
+                    // Remove the previous color css class
+                    let mut sender_color_class = self_.sender_color_class.borrow_mut();
+                    if let Some(class) = sender_color_class.as_ref() {
+                        self_.sender_label.remove_css_class(class);
+                        *sender_color_class = None;
+                    }
+
+                    // Color sender label
+                    if let MessageSender::User(user) = message.sender() {
+                        let classes = vec![
+                            "sender-text-red",
+                            "sender-text-orange",
+                            "sender-text-violet",
+                            "sender-text-green",
+                            "sender-text-cyan",
+                            "sender-text-blue",
+                            "sender-text-pink",
+                        ];
+
+                        let color_class = classes[user.id() as usize % classes.len()];
+                        self_.sender_label.add_css_class(color_class);
+
+                        *sender_color_class = Some(color_class.into());
+                    }
+
+                    self_.sender_label.set_visible(true);
+                } else {
+                    self_.sender_label.set_visible(false);
```

Comment:
Question:
Here we set the visibility to `false`. Shouldn't we explicitly set it to true again for sponsored messages in case the widget is recycled?